File: Ejercicio 013/colegio.java
class Persona {
    String nombre;
    String id;
    int edad;

    public Persona(String nombre, String id, int edad) {
        this.nombre = nombre;
        this.id = id;
        this.edad = edad;
    }

    public void mostrarInfo() {
        System.out.println("Nombre: " + nombre);
        System.out.println("ID: " + id);
        System.out.println("Edad: " + edad);
    }

    public void rol() {}
}


class Estudiante extends Persona {
    String grado;
    double promedio;

    public Estudiante(String nombre, String id, int edad, String grado, double promedio) {
        super(nombre, id, edad);
        this.grado = grado;
        this.promedio = promedio;
    }

    public void rol() {
        System.out.println("Grado: " + grado);
        System.out.println("Promedio: " + promedio);
    }
}


class Profesor extends Persona {
    String materia;
    double salario;

    public Profesor(String nombre, String id, int edad, String materia, double salario) {
        super(nombre, id, edad);
        this.materia = materia;
        this.salario = salario;
    }

    public void rol() {
        System.out.println("Materia: " + materia);
        System.out.println("Salario: " + salario);
    }
}


class Directivo extends Persona {
    String cargo;
    int experiencia;

    public Directivo(String nombre, String id, int edad, String cargo, int experiencia) {
        super(nombre, id, edad);
        this.cargo = cargo;
        this.experiencia = experiencia;
    }

    public void rol() {
        System.out.println("Cargo: " + cargo);
        System.out.println("Años de experiencia: " + experiencia);
    }
}


public class Main {
    public static void main(String[] args) {

        Persona p1 = new Estudiante("Sofia", "101", 16, "11°", 4.5);
        Persona p2 = new Profesor("Carlos", "202", 40, "Matemáticas", 2500000);
        Persona p3 = new Directivo("Marta", "303", 50, "Rectora", 20);

        Persona[] personas = {p1, p2, p3};

        for (Persona p : personas) {
            System.out.println("Persona: " + p.getClass().getSimpleName());
            p.mostrarInfo();
            p.rol();
            System.out.println("-------------------");
        }
    }
}

File: Ejercicio 013/colegio.py
class Persona:
    def __init__(self, nombre, id, edad):
        self.nombre = nombre
        self.id = id
        self.edad = edad

    def mostrar_info(self):
        print("Nombre:", self.nombre)
        print("ID:", self.id)
        print("Edad:", self.edad)

    def rol(self):
        pass


class Estudiante(Persona):
    def __init__(self, nombre, id, edad, grado, promedio):
        super().__init__(nombre, id, edad)
        self.grado = grado
        self.promedio = promedio

    def rol(self):
        print("Grado:", self.grado)
        print("Promedio:", self.promedio)


class Profesor(Persona):
    def __init__(self, nombre, id, edad, materia, salario):
        super().__init__(nombre, id, edad)
        self.materia = materia
        self.salario = salario

    def rol(self):
        print("Materia:", self.materia)
        print("Salario:", self.salario)


class Directivo(Persona):
    def __init__(self, nombre, id, edad, cargo, experiencia):
        super().__init__(nombre, id, edad)
        self.cargo = cargo
        self.experiencia = experiencia

    def rol(self):
        print("Cargo:", self.cargo)
        print("Años de experiencia:", self.experiencia)


# MAIN
p1 = Estudiante("Sofia", "101", 16, "11°", 4.5)
p2 = Profesor("Carlos", "202", 40, "Matemáticas", 2500000)
p3 = Directivo("Marta", "303", 50, "Rectora", 20)

personas = [p1, p2, p3]

for p in personas:
    print("Persona:", type(p).__name__)
    p.mostrar_info()
    p.rol()
    print("-------------------")
